### ISSUE TITLE
Request: Linux Version?

### ISSUE THREAD

**Issue by mistergibson** [TCB13/LoFloccus#4]
Do you have any plans to fully support Linux / X11 / Qt5 ?

I'd like very much to have this as a sync option, but I *only* run linux here.

**Comment by TCB13** [comment]
Your issue is very vague. Can you tell us what is not working exactly? There have been multiple users running on Ubuntu and derivatives saying it all works fine.

What's your distro, desktop env etc?

**Comment by marcelklehr** [comment]
> 1. Install Qt

Is probably something most people would rather like to avoid.

**Comment by TCB13** [comment]
@marcelklehr I'm not that experienced with packing stuff for linux... and there are so many distros, architectures etc...

**Comment by marcelklehr** [comment]
> people who are using linux are usually able to install Qt and tweak stuff around

true. [AppImage](https://docs.appimage.org/packaging-guide/index.html) is pretty neat, though... :angel:

**Comment by petko10** [comment]
The installation got too lengthy for me and I found an alternative to easily setup a WebDav share on linux (via PyWebDAV3): 
```
pip install PyWebDAV3 --user
```
And then put this as an autostart entry (maybe test it in the terminal first):
```
davserver --directory /sync/random_unique/bookmarks --user e_g_bookmarks_user --password your_password --daemon
```

I know it's kind of a workaround, and it may be less secure, but it beats not being able to use Floccus on linux easily.

**Comment by petko10** [comment]
The installation got too lengthy for me and I found an alternative to easily setup a WebDav share on linux. Install WsgiDAV via pip and then put this as an autostart entry (maybe test it in the terminal first):
```
wsgidav -r /path/to/bookmarks/folder -p 8008 --auth anonymous --server gevent
```
(I tried with the default cherry server, but encountered some error which I could not trace)

It's a workaround and not as secure as LoFloccus, but it's way easyer than the installation process and allows linux users to start with Floccus.

**Comment by BirdInFire** [comment]
> @marcelklehr I'm not that experienced with packing stuff for linux... and there are so many distros, architectures etc... And people who are using linux are usually able to install Qt and tweak stuff around.

pretty weak argument, do you know that install dev package on a system can make it unstable or even break the interface in the worst cases, so using the argument of "linuxians know" is risky, moreover many "linuxians" are former Windows users who have never opened a terminal in their life ...

**Comment by TCB13** [comment]
> pretty weak argument, do you know that install dev package on a system can make it unstable or even break the interface in the worst cases, so using the argument of "linuxians know" is risky, moreover many "linuxians" are former Windows users who have never opened a terminal in their life ...

C'mon, I re-wrote this project from the scratch in order to have less dependencies and work way better under Windows/macOS a few weeks ago. The door for a Linux version is open but it might take some time.

**Comment by joaoherberto** [comment]
+1 for Linux

**Comment by sunjam** [comment]
> The door for a Linux version is open but it might take some time.

That would be fantastic.



> Your issue is very vague. Can you tell us what is not working exactly? There have been multiple users running on Ubuntu and derivatives saying it all works fine.

On Ubuntu 22.04.1  and Linux users are hoping to install prepackaged releases, no different than Windows and Mac users.  There are a couple excellent, portable packaging formats for Linux: [AppImage](https://appimage.org/) and [Flatpak](https://docs.flatpak.org/en/latest/).

Either packaging can be installed on any Linux system.  Thanks for considering.  Awesome project.

**Comment by TCB13** [comment]
> > The door for a Linux version is open but it might take some time.
> 
> That would be fantastic.
> 
> > Your issue is very vague. Can you tell us what is not working exactly? There have been multiple users running on Ubuntu and derivatives saying it all works fine.
> 
> On Ubuntu 22.04.1 and Linux users are hoping to install prepackaged releases, no different than Windows and Mac users. There are a couple excellent, portable packaging formats for Linux: [AppImage](https://appimage.org/) and [Flatpak](https://docs.flatpak.org/en/latest/).
> 
> Either packaging can be installed on any Linux system. Thanks for considering. Awesome project.

Are you able to PR a working version packaged in one of those formats? Not that I particularly like them but if someone can do it I will be happy to accept contributions and link to them.

Thank you.

**Comment by userdocs** [comment]
Alpine linux is the way to go for a fully static builds as it is

- 1: easy to do and 
- 2: uses musl not glibc (which is what debian types uses and makes it hard)

If this could be done via alpine, it would be good.

ideally headless, as in no need to install the desktop, but just libs it needs.

I looked at the install, i got as far as making the go stuff in libs, but then not really sure what to do on a headless system to build the app.

They also recently add qt creator 

https://pkgs.alpinelinux.org/packages?name=qt-creator&branch=edge&repo=&arch=&maintainer=

I think this would a good path to explore. so you end up with 3 static binaries.

**Comment by TCB13** [comment]
@userdocs there's no point in an "headless" version, you can simply run nginx with webdav support and you're done. The entire point of the project is to provide a desktop / GUI experience.

**Comment by userdocs** [comment]
@TCB13 that's specific to building the app. Alpine has a desktop. If you wanted a fully static linux binary that will run on any other linux based distro, Alpine is the way to do it.

**Comment by debilin** [comment]
+1 for Linux static build
